### PR TITLE
New Spreadsheet Mod

### PR DESCRIPTION
**Pull request by LFDPhoenix1411**
**Please describe the changes this PR makes and why it should be merged:**

Added Spreadsheet Mod

Merging this PR will allow Discord bots to read CSV files, providing a lightweight and efficient way to access static datasets. Furthermore, working with CSV files is often easier for new users than using SQL, making it more accessible for users who are not familiar with complex database systems.

**Status**

- [x] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [x] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by TheMonDon** [comment]
You need to add the mod to the docs as well and probably will have to run prettier

**Comment by LFDPhoenix1411** [comment]
Edited the Docs to add spreadsheet_MOD.js
run prettier, but I'm not sure if it changed anything

**Comment by OneAndonlyFinbar** [comment]
Can someone test the action? Also fix eslint please

**Comment by OneAndonlyFinbar** [comment]
@TheMonDon

**Comment by LFDPhoenix1411** [comment]
im gonna fix the ESLint

**Comment by TheMonDon** [comment]
> im gonna fix the ESLint

You need to do `npm run prettier:fix`

I will review the mod after you fix prettier so I don't have to deal with that

**Comment by LFDPhoenix1411** [comment]
Hopefully it works now, couldn't use `npm run prettier:fix`

File: actions/spreadsheet_MOD.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    // get a row with the help with a key and a value
+    function getRow(filePath, columnName, columnValue, callback) {
+      const dataSheet = [];
+      const parser = fs.createReadStream(filePath).pipe(
+        parse({
+          delimiter: ',',
+          columns: true,
+          ltrim: true,
+        })
```
Comment (DetroitWhiskey136) [inline actions/spreadsheet_MOD.js:73]:
```suggestion
        }),
```

File: actions/spreadsheet_MOD.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    `;
+  },
+
+  init() {},
+
+  async action(cache) {
+    // get packages and data
+    const data = cache.actions[cache.index];
+    const fs = require('fs');
+    const { parse } = require('csv-parse');
```
Comment (TheMonDon) [inline actions/spreadsheet_MOD.js:59]:
```suggestion
    const Mods = this.getMods();
    const { parse } = Mods.require('csv-parse');
```

**Comment by TheMonDon** [comment]
It looks good, seems to work. It might be beneficial to add text saying exactly what the mod does. The action line comment saying row is helpful though.

**Review by TheMonDon — APPROVED** [review]

**Review by DetroitWhiskey136 — APPROVED** [review]

**Pull request merged**